cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractInMemoryProgramRunner.java

```diff
@@ -80,7 +80,8 @@ public AbstractInMemoryProgramRunner(CConfiguration cConf, ProgramRunnerFactory
   private void startComponent(Program program, String name, int instances, RunId runId, ProgramOptions options,
                                 Table<String, Integer, ProgramController> components, ProgramRunnerFactory.Type type) {
     for (int instanceId = 0; instanceId < instances; instanceId++) {
-      ProgramOptions componentOptions = createComponentOptions(name, instanceId, instances, runId, options);
+      ProgramOptions componentOptions = createComponentOptions(name + "-" + instanceId,
```

Comment:
Maybe better to refactor the existing "createComponentOptions" method as well (current change is to just fit in what the current logic is, since the name is being passed as component name to the program controller).